hostapd: update hostapd config

BigNerd95 opened the pull request:
Update the config file to the latest version
Enabled flags are the same as before

Tested compiling and comparing the md5sum of the new generated binary with the md5sum of the previous binary
    
Signed-off-by: Lorenzo Santina <lorenzo.santina@edu.unito.it>

devianceluka commented:
This adds bss_transition?

devianceluka commented:
+# Wireless Network Management (IEEE Std 802.11v-2011)
+# Note: This is experimental and not complete implementation.
+#CONFIG_WNM=y
Does this add bss_transition?

BigNerd95 commented:
Uncommenting "#CONFIG_WNM=y" and updating hostapd.sh yes
I'll make this change in a future commit
Now I'm updating wpa_supplicant config

BigNerd95 commented:
If you have patience I'm adding bss_transition support in another commit

BigNerd95 commented:
I'm only waiting that someone accept this PR

devianceluka commented:
Please add it!

BigNerd95 commented:
I'll add it

BigNerd95 commented:
IEEE 802.11v support added to hostapd-full and wpa_supplicant-full

devianceluka commented:
Bravo!!!

devianceluka commented:
Can I do anything to get this in faster?

devianceluka commented:
The only thing now missing, when we have 802.11r+v, is band steering

devianceluka commented:
"Neighbor table" section of hostapd.conf:
# Reject authentication from a station that was seen on another radio.
# Default: Disabled
#
# This can be used with enabled track_sta_max_num configuration on another
# interface controlled by the same hostapd process to reject authentication
# attempts from a station that has been detected to be capable of operating on
# another band, e.g., to try to reduce likelihood of the station selecting a
# 2.4 GHz BSS when the AP operates both a 2.4 GHz and 5 GHz BSS concurrently.
#
# Note: Enabling this can cause connectivity issues and increase latency for
# connecting with the AP.
#no_auth_if_seen_on=wlan1

devianceluka commented:
"Neighbor table" section of hostapd.conf:

# Reject authentication from a station that was seen on another radio.
# Default: Disabled
#
# This can be used with enabled track_sta_max_num configuration on another
# interface controlled by the same hostapd process to reject authentication
# attempts from a station that has been detected to be capable of operating on
# another band, e.g., to try to reduce likelihood of the station selecting a
# 2.4 GHz BSS when the AP operates both a 2.4 GHz and 5 GHz BSS concurrently.
#
# Note: Enabling this can cause connectivity issues and increase latency for
# connecting with the AP.
#no_auth_if_seen_on=wlan1

devianceluka commented:
`Neighbor table` section of https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf

> \# Reject authentication from a station that was seen on another radio.
\# Default: Disabled
\#
\# This can be used with enabled track_sta_max_num configuration on another
\# interface controlled by the same hostapd process to reject authentication
\# attempts from a station that has been detected to be capable of operating on
\# another band, e.g., to try to reduce likelihood of the station selecting a
\# 2.4 GHz BSS when the AP operates both a 2.4 GHz and 5 GHz BSS concurrently.
\#
\# Note: Enabling this can cause connectivity issues and increase latency for
\# connecting with the AP.
\#no_auth_if_seen_on=wlan1

stintel commented:
As for 802.11v, the implementation is experimental and incomplete. Do we really want to enable it? If it's enabled in the config, is it configurable at runtime? If it isn't, it might be safer not to enable it. If it is, please add UCI support for it. Or just drop the commit from this PR so we can merge this one, and discuss 802.11v in a new PR.

BigNerd95 commented:
Ok, I'll fix the commit message and I'll split wpa_supplicant commit in 3 commits

I commented CONFIG_IEEE80211W=y in wpa_supplicant-full.config because it is set in the Makefile, only if the driver support it (in the same way as of hostapd)

I added CONFIG_EAP_FAST=y in wpa_supplicant-p2p.config because it was the only missing flag about EAP compared to wpa_supplicant-full.config, if it is a problem I can comment it

I removed NEED_80211_COMMON because it is not part of config file, it is set by the hostapd upstream Makefile

About 802.11v I already added the support for UCI
With this commit it only add the support at compile time by default, but if you want to use it, you have to add ieee80211v=1 in /etc/config/wireless and then specify the flags for bss_transition, etc.

BigNerd95 commented:
Fixed commits

BigNerd95 commented:
What do you think about removing the debug from mini versions?
It reduce considerably the binary size

It can be done in two way:
- uncommenting CONFIG_NO_STDOUT_DEBUG=y
- setting this flag via menuconfig (more versatile)

stintel commented on package/network/services/hostapd/files/hostapd.sh:
```diff
@@ -376,6 +380,27 @@ hostapd_set_bss_options() {
+		append bss_conf "wnm_sleep_mode=$wnm_sleep_mode" "$N"
+		append bss_conf "bss_transition=$bss_transition" "$N"
+		#append bss_conf "proxy_arp=$proxy_arp" "$N"
+		#append bss_conf "na_mcast_to_ucast=$na_mcast_to_ucast" "$N"
```
If you're going to comment them out, please leave them out entirely.

BigNerd95 replied:
Ok, I commented them out because the hostapd version (2016-12-19) is not updated and those flags were not yet implemented in that versoin.
But I've just finished updating the hostapd package to the latest upstream commit on another branch and I think I'll make a PR about the hostapd update first and then I'll make the 80211v PR.

stintel replied:
Don't put too much effort in updating hostapd. I have an update queued in my staging dir for some weeks now, but it breaks build due to some conflicts. If you do want to make an update, please start from https://git.lede-project.org/?p=lede/stintel/staging.git;a=shortlog;h=refs/heads/hostapd

BigNerd95 replied:
What kind of conflicts?
I've fixed the patches errors using quilt and mine is compiling fine.
Where we can discuss about this?

stintel replied:
See http://lists.infradead.org/pipermail/lede-dev/2017-September/008788.html
You can reply to the thread, or come on IRC, or ... Either way, since I did not get any feedback, I'm probably going to merge those changes somewhere this week.

BigNerd95 replied:
```
since I did not get any feedback
```
Did you see this?
http://lists.infradead.org/pipermail/lede-dev/2017-September/008799.html

stintel replied:
Yes, you misunderstood. That's the patch that is in my staging tree. I did not get feedback about the 2 patches to fix the build problem.

stintel commented on package/network/services/hostapd/files/hostapd.sh:
```diff
@@ -376,6 +380,27 @@ hostapd_set_bss_options() {
+		set_default na_mcast_to_ucast 0
+
+		append bss_conf "time_advertisement=$time_advertisement" "$N"
+		[ -n "$time_zone" ] && append bss_conf "time_zone=$time_zone" "$N"
```
Maybe we should get this from timezone in /etc/config/system instead of setting the same setting in different locations?

BigNerd95 replied:
hostapd.conf
```
# Local time zone as specified in 8.3 of IEEE Std 1003.1-2004:
# stdoffset[dst[offset][,start[/time],end[/time]]]
#time_zone=EST5
```
/etc/config/system
```
option timezone 'UTC'
```
Yes, the format looks like the same.
I think we can discuss better about this in the dedicated PR.

stintel replied:
I had already checked. Mine is ```option timezone 'EET-2EEST,M3.5.0/3,M10.5.0/4'``

stintel commented:
The same goes for CONFIG_NO_STDOUT_DEBUG - put it in a different PR so we can discuss/review there.

stintel commented:
Can you just drop the last commit from this PR, so I can merge it?

BigNerd95 commented:
Done :-)

stintel commented:
Pulled into my staging tree: https://git.lede-project.org/?p=lede/stintel/staging.git
